code/src/components/Survey.js

```diff
@@ -0,0 +1,207 @@
+import React, { useState } from 'react'
+
+import Summary from './Summary'
+import InputRadioButton from 'components/InputRadioButton'
+import Button from 'components/Button'
+import InputDropDown from 'components/InputDropDown'
+import InputText from 'components/InputText'
+import InputCheckBox from 'components/InputCheckBox'
+import InputCheckBoxGroup from 'components/InputCheckBoxGroup'
+
+const Survey = () => {
+  
+  //STATES
+
+  const [question, setQuestion] = useState(0)
+  const [enjoyReading, setEnjoyReading] = useState('')
+  const [readingFormat, setReadingFormat] = useState('')
+  const [book, setBook] = useState ('')
+  const [isChecked, setIsChecked] = useState(false)
+  const [checkboxGroup, setCheckboxGroup] = useState([])
+  const [showSummary, setShowSummary] = useState(false)
+
+  //VARIABLES
+
+  const enjoyReadingArray = ['I enjoy reading a lot', 'I like reading sometimes', 'I do not like reading']
+  const checkboxGroupArray = ['Adventure', 'Classics', 'Detective', 'Fantasy', 'History', 'Romance', 'Science Fiction', 'Poetry', 'Self-Help', 'Other']
+
+  //FUNCTIONS
+
+  // This function leads to the next question
+  const onNextQuestionChange = () => {
+    setQuestion(question + 1)
+  }
+
+  // This function goes back to the previous question
+  const onPreviousQuestionChange = () => setQuestion(question - 1)
+
+  //This function will show the summary of answers and prevet the page from refreshing on submit form
+  const handleSubmit = event => {
+    event.preventDefault()
+    setShowSummary(true)
+  }
+  
+  // This function stores the answer from the first question (radio input)
+  const onEnjoyReadingChange = event => setEnjoyReading(event.target.value)
+
+  // This function stores the answer from the second question (select input)
+  const onReadingFormatChange = event => setReadingFormat(event.target.value)
+
+  // This function stores the answer from the third question (text input)
+  const onBookChange = event => setBook(event.target.value)
+
+  // This function stores the answer from the fourth question (checkbox input)
+  const onIsCheckedChange = event => setIsChecked(event.target.checked)
+
+  // This function will update the value of an array/answer for the fifth question (checkbox group input)
```

Comment:
This structure with it's comment is heaven, Irina! 😇 It makes it so easy to understand your code, that even I can understand the code in an acceptable amount of time - so thanks for that! 🤓 🙌 🥳